android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -767,10 +767,11 @@ abstract class EngineSession(
      * will be restored.
      *
      * @param state A saved session state.
+     * @param enableDesktopMode True, if desktop mode option is enabled, false otherwise
      * @return true if the engine session has successfully been restored with the provided state,
      * false otherwise.
      */
-    abstract fun restoreState(state: EngineSessionState): Boolean
+    abstract fun restoreState(state: EngineSessionState, enableDesktopMode: Boolean): Boolean
```

Comment:
We shouldn't be passing the flag individually as a param here. What happens when we want to add more things? That shouldn't mean n parameters.

We should get the toggle state from the `EngineSessionState` which each of the engine's implement.